From: Stanislav Brabec <sbrabec@suse.cz>
Date: Tue, 20 Nov 2018 00:38:14 +0100
Subject: agetty: Return old behavior with empty logname

c094fcd37 introduced a behavior change: When Return is entered with empty
logname, nothing happens. As it confuses users, return back the old
behavior: re-prompt.

Signed-off-by: Stanislav Brabec <sbrabec@suse.cz>
---
 term-utils/agetty.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/term-utils/agetty.c b/term-utils/agetty.c
index 759cea5..7c11c9c 100644
--- a/term-utils/agetty.c
+++ b/term-utils/agetty.c
@@ -2185,7 +2185,7 @@ static char *get_logname(struct issue *ie, struct options *op, struct termios *t
 				break;
 			}
 			/* Everything was erased. */
-			if (bp == logname)
+			if (bp == logname && cp->eol == '\0')
 				goto no_reload;
 		}
 	}
